### PR TITLE
feat: use FallbackProvider in the DTL

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Replaces `StaticJsonRpcProvider` with `FallbackProvider` so we can run multiple nodes and get a consensus of results from each.

### CODE REVIEW

**Comment by tynes** [comment]
This needs a changeset still.

**Comment by tynes** [comment]
CI is generally failing here, need to investigate as to why

**Comment by tynes** [comment]
cc @smartcontracts this is ready for review again

File: packages/core-utils/src/provider.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+// Create a fallback provider using a comma delimited
+// string of URLs
+export const FallbackProvider = (config: string) => {
+  const providers = []
+  const urls = config.split(',')
+  for (const url of urls) {
+    const provider = new ethers.providers.StaticJsonRpcProvider(url)
+    providers.push(provider)
+  }
+  return new ethers.providers.FallbackProvider(providers)
```
Comment (K-Ho) [inline packages/core-utils/src/provider.ts:16]: looks good - note here that we can further parameterize things like weights of different providers by providing an array of  `FallbackProviderConfig` instead.
> FallbackProviderConfig. If a Provider is provided, the defaults are a priority of 1 and a weight of 1.

**Review by K-Ho — APPROVED** [review]

File: packages/core-utils/src/provider.ts (hunk 1 above)
Comment (tynes) [reply packages/core-utils/src/provider.ts:16]: https://github.com/ethers-io/ethers.js/blob/c2c0ce75039e7256b287f9a764188d08ed0b7296/packages/providers/src.ts/fallback-provider.ts#L97

The existing implementation will randomize which is used. We could set the priority based on the order of URLs passed in

Comment (smartcontracts) [reply packages/core-utils/src/provider.ts:16]: Does this do quorum by default?

File: packages/core-utils/src/provider.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/**
+ * Provider Utilities
+ */
+
+import { ethers } from 'ethers'
+import { Provider } from '@ethersproject/providers'
+
+// Copied from @ethersproject/providers since it is not
+// currently exported
```
Comment (tynes) [inline packages/core-utils/src/provider.ts:9]: PR to export the interface here: https://github.com/ethers-io/ethers.js/pull/2121

**Pull request merged**